execution-engine/contracts/hdac-system/pop/src/contract_queue/request_queue.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::{boxed::Box, vec::Vec};
+use core::result;
+
+use types::{
+    bytesrepr::{self, FromBytes, ToBytes},
+    system_contract_errors::pos::{Error, Result},
+    BlockTime, CLType, CLTyped, U512,
+};
+
+#[derive(Default, PartialEq)]
+pub struct RequestQueue<T: RequestKey>(pub Vec<RequestQueueEntry<T>>);
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct RequestQueueEntry<T: RequestKey> {
+    pub request_key: T,
+    pub amount: U512,
+    pub timestamp: BlockTime,
```

Comment:
Since we use the `ulb` concept, wouldn't the `block height` fit more than the `block time`?

Reply:
As far as I know, `ulb` is the concept about the height allowing us to assure it must have been finalized. But the undelegation delay serves as an economic variable or duration for slashing.
So I thought it works in a time concept such as 1 day, 10 days, etc.

I am curious more specifically why do you think the block height is fitter than time.

And I think this needs to be discussed in our team channel.